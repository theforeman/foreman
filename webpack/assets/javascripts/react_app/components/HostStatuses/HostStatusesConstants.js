export const GLOBAL_STATUS_OK = 0;
export const GLOBAL_STATUS_WARN = 1;
export const GLOBAL_STATUS_ERROR = 2;
export const HOST_STATUSES_KEY = 'HOST_STATUSES';
export const API_OPTIONS = { key: HOST_STATUSES_KEY };
