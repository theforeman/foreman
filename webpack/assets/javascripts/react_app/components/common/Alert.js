import React from 'react';
import Icon from './Icon';
import { ALERT_CSS } from '../../constants';

const Alert = (props) => {
  let classNames = ALERT_CSS[props.type];

  if (props.css) {
    classNames += ' ' + props.css;
  }

  if (props.dismissable) {
    classNames += ' ' + ALERT_CSS.dismissable;
  }

  const closeBtn = (
    <button className="close" data-dismiss="alert" aria-hidden="true">
      <Icon type="close" />
    </button>
  );

  return (
    <div className={classNames}>
      {props.dismissable && closeBtn}
      {props.children}
    </div>
  );
};

Alert.defaultProps = { dismissable: false, css: null, type: 'info' };
export default Alert;
