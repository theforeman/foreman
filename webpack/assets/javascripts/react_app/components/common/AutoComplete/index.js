import React, { Component } from 'react';
import Downshift from 'downshift';
import { Dropdown, MenuItem, InputGroup, FormControl, Button } from 'patternfly-react';
import PropTypes from 'prop-types';

export const renderItems = ({
  items, highlightedIndex, selectedItem, getItemProps,
}) => (
  <Dropdown.Menu
    style={{
      display: 'block',
      position: 'absolute',
      left: 'inherit',
      marginTop: 0,
      top: 'auto',
    }}
  >
    {items.map((item, index) => (
      <MenuItem
        {...getItemProps({
          index,
          item,
          active: highlightedIndex === index,
          onClick: (e) => {
            // At this point the event.defaultPrevented
            // is already set to true by react-bootstrap
            // MenuItem. We need to set it back to false
            // So downshift will execute it's own handler
            e.defaultPrevented = false;
          },
        })}
        key={item}
      >
        {item}
      </MenuItem>
    ))}
  </Dropdown.Menu>
);

class AutoComplete extends Component {
  constructor(props) {
    super(props);

    this.state = {
      inputValue: '',
    };
  }

  render() {
    const {
      onSearch, labelText, onInputUpdate, items, actionText, ...rest
    } = this.props;

    return (
      <Downshift
        onStateChange={({ inputValue }) => {
          if (typeof inputValue === 'string') {
            onInputUpdate(inputValue);
            this.setState({ inputValue });
          }
        }}
        defaultHighlightedIndex={0}
        selectedItem={this.state.inputValue}
        {...rest}
      >
        {({
          getInputProps,
          getItemProps,
          getLabelProps,
          isOpen,
          inputValue,
          highlightedIndex,
          getRootProps,
          selectedItem,
        }) => (
          <div>
            {labelText && <label {...getLabelProps()}>{labelText}</label>}
            <InputGroup>
              <FormControl type="text" {...getInputProps()} />
              <InputGroup.Button>
                <Button onClick={() => onSearch(inputValue)}>{actionText}</Button>
              </InputGroup.Button>
            </InputGroup>

            {isOpen && items.length
              ? renderItems({
                  items,
                  highlightedIndex,
                  selectedItem,
                  getItemProps,
                })
              : null}
          </div>
        )}
      </Downshift>
    );
  }
}

AutoComplete.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
  onInputUpdate: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  labelText: PropTypes.string,
  actionText: PropTypes.string,
};

AutoComplete.defaultProps = {
  labelText: '',
  actionText: 'Search',
};

export default AutoComplete;
