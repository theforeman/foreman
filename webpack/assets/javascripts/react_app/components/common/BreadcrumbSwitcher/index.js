import React from 'react';
import {
  Popover,
  Button,
  Icon,
  ListGroup,
  ListGroupItem,
  OverlayTrigger,
} from 'patternfly-react';
import { ButtonToolbar } from 'react-bootstrap';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
import PropTypes from 'prop-types';
import './BreadcrumbSwitcher.scss';

const BreadcrumbSwitcher = ({ resources }) => {
  const popover = (
    <Popover id="popover-contained">
      <ListGroup className="scrollable-list">
        {resources.map((item) => {
          const { onClick, url, caption } = item;
          return (
            <ListGroupItem
              className="no-border"
              key={`id-${caption}`}
              href={url}
              onClick={onClick}
            >
              <EllipsisWithTooltip>{caption}</EllipsisWithTooltip>
            </ListGroupItem>
          );
        })}
      </ListGroup>
    </Popover>
  );
  return (
    <ButtonToolbar>
      <OverlayTrigger
        rootClose
        trigger="click"
        placement="bottom"
        overlay={popover}
      >
        <Button>
          <Icon type="fa" name="exchange" />
        </Button>
      </OverlayTrigger>
    </ButtonToolbar>
  );
};

BreadcrumbSwitcher.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  caption: PropTypes.string,
};

export default BreadcrumbSwitcher;
