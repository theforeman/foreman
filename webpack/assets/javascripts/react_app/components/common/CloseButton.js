import React from 'react';
import Icon from './Icon';

const CloseButton = ({onClick}) => {
  return (
    <button className="close" aria-hidden="true" onClick={onClick}>
      <Icon type="close" />
    </button>
  );
};

CloseButton.propTypes = {
  onClick: React.PropTypes.func.isRequired
};

export default CloseButton;
