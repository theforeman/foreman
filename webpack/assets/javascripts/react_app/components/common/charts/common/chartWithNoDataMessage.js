import React from 'react';
import MessageBox from '../../MessageBox';

export const chartWithNoDataMessage = (WrappedComp, noDataMsg = 'No data available') => (props) => {
  const columns = props && props.data && props.data.columns ? props.data.columns : [];
  if (columns.length > 0) {
    return (<WrappedComp {...props} />);
  }
  return <MessageBox msg={noDataMsg} icontype="info" />;
};
