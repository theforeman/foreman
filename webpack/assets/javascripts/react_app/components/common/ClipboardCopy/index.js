import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, OverlayTrigger, Tooltip } from 'patternfly-react';
import UUID from 'uuid/v1';
import { translate as __ } from '../../../common/I18n';
import './clipboard-copy.scss';

const ClipboardCopy = ({
  text: defaultText,
  successMessage,
  buttonText,
  textareaProps,
  buttonProps,
}) => {
  const [text, setText] = useState(defaultText);

  return (
    <div className="clipboard-copy">
      <textarea
        defaultValue={text}
        onChange={({ target: { value } }) => setText(value)}
        {...textareaProps}
      />
      <OverlayTrigger
        overlay={<Tooltip id={UUID()}>{successMessage}</Tooltip>}
        placement="right"
        trigger={['click']}
        rootClose
      >
        <Button
          onClick={() => navigator.clipboard.writeText(text)}
          bsStyle="default"
          {...buttonProps}
        >
          {buttonText}
        </Button>
      </OverlayTrigger>
    </div>
  );
};

ClipboardCopy.propTypes = {
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
  successMessage: PropTypes.string,
  textareaProps: PropTypes.object,
  buttonProps: PropTypes.object,
};

ClipboardCopy.defaultProps = {
  buttonText: __('Copy to clipboard'),
  successMessage: __('Copied!'),
  textareaProps: {},
  buttonProps: {},
};

export default ClipboardCopy;
