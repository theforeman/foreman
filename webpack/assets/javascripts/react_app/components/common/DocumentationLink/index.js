import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem, Icon } from 'patternfly-react';
import { newWindowOnClick } from '../../../common/helpers';
import { translate as __ } from '../../../../react_app/common/I18n';

const DocumentationLink = ({ href, children }) => (
  <MenuItem key="documentationUrl" href={href} onClick={newWindowOnClick(href)}>
    <Icon type="fa" name="question-circle" />
    {` ${children}`}
  </MenuItem>
);

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node,
};

DocumentationLink.defaultProps = {
  children: __('Documentation'),
};

export default DocumentationLink;
