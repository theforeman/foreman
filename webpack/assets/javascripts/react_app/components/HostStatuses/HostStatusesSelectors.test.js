import { testSelectorsSnapshotWithFixtures } from '../../common/testHelpers';
import { get } from 'lodash';
import { HOST_STATUSES_KEY } from './HostStatusesConstants';
import {
  selectHostStatusesNames,
  selectGlobalStatus,
  selectHostStatusDetails,
  selectHostStatusDescription,
  selectHostStatusCounter,
  selectHostStatusTotalPaths,
  selectHostStatusOwnedPaths
} from './HostStatusesSelectors';
import { store } from './HostStatuses.fixtures.js'

const statusName = get(store, ['API', HOST_STATUSES_KEY, 'response', 'results', '0', 'name']);

const fixtures = {
  'selects HostStatusesNames': () => selectHostStatusesNames(store),
  'selects HostStatusDetails': () => selectHostStatusDetails(store, statusName),
  'selects HostStatusDescription': () => selectHostStatusDescription(store, statusName),
  'selects selectHostStatusTotalPaths': () => selectHostStatusTotalPaths(store, statusName),
  'selects selectHostStatusOwnedPaths': () => selectHostStatusOwnedPaths(store, statusName),
  'selects GlobalStatus': () => selectGlobalStatus(store, statusName),
  'selects HostStatusCounter': () => selectHostStatusCounter(store, statusName)
};

describe('HostStatusesSelectors selectors', () => {
  testSelectorsSnapshotWithFixtures(fixtures);
});
