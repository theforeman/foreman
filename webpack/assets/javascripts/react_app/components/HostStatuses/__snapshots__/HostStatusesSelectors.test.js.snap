// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostStatusesSelectors selectors selects GlobalStatus 1`] = `2`;

exports[`HostStatusesSelectors selectors selects HostStatusCounter 1`] = `
Object {
  "error": Object {
    "owned": 0,
    "total": 5,
  },
  "ok": Object {
    "owned": 1,
    "total": 3,
  },
  "unknown": Object {
    "owned": 0,
    "total": 0,
  },
  "warn": Object {
    "owned": 2,
    "total": 7,
  },
}
`;

exports[`HostStatusesSelectors selectors selects HostStatusDescription 1`] = `"Description of the status"`;

exports[`HostStatusesSelectors selectors selects HostStatusDetails 1`] = `
Array [
  Object {
    "global_status": 0,
    "label": "OK",
    "owned": 1,
    "owned_path": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+ok",
    "total": 3,
    "total_path": "/hosts?search=status+%3D+ok",
  },
  Object {
    "global_status": 1,
    "label": "Warning",
    "owned": 2,
    "owned_path": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+warn",
    "total": 7,
    "total_path": "/hosts?search=status+%3D+warn",
  },
  Object {
    "global_status": 2,
    "label": "Error",
    "owned": 0,
    "owned_path": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+error",
    "total": 5,
    "total_path": "/hosts?search=status+%3D+error",
  },
]
`;

exports[`HostStatusesSelectors selectors selects HostStatusesNames 1`] = `
Array [
  "Status Name",
]
`;

exports[`HostStatusesSelectors selectors selects selectHostStatusOwnedPaths 1`] = `
Object {
  "errorOwnedPath": "/hosts?search=status+%3D+error",
  "okOwnedPath": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+ok",
  "warnOwnedPath": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+warn",
}
`;

exports[`HostStatusesSelectors selectors selects selectHostStatusTotalPaths 1`] = `
Object {
  "errorTotalPath": "/hosts?search=owner+%3D+current_user+AND+%28status+%3D+error",
  "okTotalPath": "/hosts?search=status+%3D+ok",
  "warnTotalPath": "/hosts?search=status+%3D+warn",
}
`;
